sentry-android-replay/src/main/java/io/sentry/android/replay/video/SimpleVideoEncoder.kt

```diff
@@ -0,0 +1,167 @@
+/**
+ * Adapted from https://github.com/fzyzcjy/flutter_screen_recorder/blob/master/packages/fast_screen_recorder/android/src/main/kotlin/com/cjy/fast_screen_recorder/SimpleVideoEncoder.kt
+ *
+ * Copyright (c) 2021 fzyzcjy
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * In addition to the standard MIT license, this library requires the following:
+ * The recorder itself only saves data on user's phone locally, thus it does not have any privacy problem.
+ * However, if you are going to get the records out of the local storage (e.g. upload the records to your server),
+ * please explicitly ask the user for permission, and promise to only use the records to debug your app.
+ * This is a part of the license of this library.
+ */
+package io.sentry.android.replay.video
+
+import android.graphics.Bitmap
+import android.media.MediaCodec
+import android.media.MediaCodecInfo
+import android.media.MediaFormat
+import android.os.Handler
+import android.os.Looper
+import android.view.Surface
+import java.io.File
+
+internal class SimpleVideoEncoder(
```

Comment:
@vaind I'd imagine you want this public right? Your idea is to take screenshots in flutter but then use the native video encoder to create a video?

Reply:
Yes, although the exact API draw API that would have the least overhead is TBD. But the DrawBitmap looks like a good start, I'm just not sure I can make it work without an additional mem copy. I've noticed there's DrawPicture but I'm not sure that would be any better.

Reply:
Ok, so the idea how we wanna do it is:
* Take screenshots every second and write them to disk
* Have a file observer or something that tracks when there are 5 screenshots (=5 seconds), then it:
  * Encodes a 5sec video
  * Creates a replay envelope and sends it

I wonder if could piggy-back on that behavior in flutter? Probably envelope creation should still be on your side because you want to pass the event through processors, collect flutter breadcrumbs etc etc, but other parts can be certainly reused. 

basically it'll be `$options.cacheDir/replay` where you'd store screenshots and then when we create a video, and we could give you a callback  like `onVideoEncoded(segmentId, bytes)` that you could use to create envelopes on your side. wdyt?

Reply:
> Take screenshots every second and write them to disk

Any reason to do this instead of encoding directly to file while the screenshot is already in memory? IO is slow and costly

Reply:
yeah, at the very least - native crashes and ANRs where we'd lose the last segment if we were to keep it in memory. @brustolin or @bruno-garcia might give more details, I think I was absent when they've implemented that

Reply:
I have no clue how the video encoder works, but if you can flush it to disk incrementally, than you could do that after each frame. But maybe that's not an option because then it actually cannot encode efficiently? 🤷

Reply:
yeah, I don't think that's an option, the video would be corrupted if you don't call `encoder.stop` or something from what I recall. but I can try that out

Reply:
if we have a safe way to write to disk on case of a crash, we could keep it in memory. We'd lose the frames in case the app was killed by the user though

Reply:
only for crashes, but native crashes and ANRs we don't have signals for to safely finish encoding, so I guess we'd have to continously store them to disk and encode a video on demand

Reply:
created a task under SDK Common just not to forget https://github.com/getsentry/sentry/issues/63255